src/services/share-buttons.service.ts

```diff
@@ -1,9 +1,7 @@
 import { Injectable, EventEmitter } from '@angular/core';
 import { Http, Jsonp, Headers, RequestOptions } from '@angular/http';
 
-import { Observable } from 'rxjs/Observable';
-import 'rxjs/add/operator/catch';
-import 'rxjs/add/Observable/empty';
+import { Observable } from 'rxjs';
```

Comment:
Well, not quite... 

First of all **Tree Shaking** has nothing to do with angular per se, it is rather an optimization provided by some bundlers like **RollupJS** or **Webpack 2**, ...

Second, it applies to the final bundled js file, which we don't have (yet) for `ng2-sharebuttons`
because we are releasing as is, all `*.js` files generated by the angular typescript compiler (`ngc`) from the source `*.ts` files.

On the other hand, this may have an impact on final users, if they don't have the proper build tools/chain when building their production code.

So not importing the whole library, specially one as big as `rxjs` is better